modules/sugar/src/main/ts/ephox/sugar/api/node/ShadowDom.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  Document, Element as DomElement,
+  HTMLElement, HTMLElementTagNameMap,
+  Node as DomNode,
+  ShadowRoot
+} from '@ephox/dom-globals';
+import * as Node from './Node';
+import * as Head from './Head';
+import * as Body from './Body';
+import { Option, Type } from '@ephox/katamari';
+import Element from './Element';
+import * as Traverse from '../search/Traverse';
+
+export type RootNode = Element<Document | ShadowRoot>;
+
+/**
+ * Is the element a ShadowRoot?
+ *
+ * Note: this is insufficient to test if any element is a shadow root, but it is sufficient to differentiate between
+ * a Document and a ShadowRoot.
+ */
+export const isShadowRoot = (dos: RootNode): dos is Element<ShadowRoot> =>
+  Node.isDocumentFragment(dos);
+
+/**
+ * Does the browser support shadow DOM?
+ *
+ * NOTE: Node.getRootNode() and Element.attachShadow don't exist on IE11 and pre-Chromium Edge.
+ */
+export const isSupported = (): boolean =>
```

Comment:
It's a trade of a teensy bit of size/performance against a teensy bit of side effect in module loading. I wasn't sure where to land on this. Should module loading be pure, or do we not care?

Reply:
When it comes to browser differences, I'm all for making them a side effect rather than constantly check a condition that never changes at runtime. It's caused occasional issues but in general I think it helps browsers optimise our performance.

Reply:
👍  Done.